lib/TAP/Formatter/Session/TeamCity.pm

```diff
@@ -206,24 +213,58 @@ sub _handle_unknown {
         $self->_finish_test('Skipped');
         $self->_finish_suite;
     }
+
+    # I'm not sure how this could ever happen, but it seems like it can under
+    # Test::Class::Moose. The "Looks like you failed ..."  message should only
+    # happen when a process exits, not when a subtest finishes.
+    elsif ( $raw =~ /^\s*# Looks like you failed \d+/ ) {
```

Comment:
In some places you've used \d+ and some places [0-9]+.  It sure would be nice to be consistent.  (The latter is probably right, given unicode semantics)